homeassistant/components/reolink/__init__.py

```diff
@@ -131,17 +131,15 @@ async def async_check_firmware_update() -> None:
         update_method=async_check_firmware_update,
         update_interval=FIRMWARE_UPDATE_INTERVAL,
     )
+
+    # If camera WAN blocked, firmware check fails and takes long, do not prevent setup
+    config_entry.async_create_task(hass, firmware_coordinator.async_refresh())
```

Comment:
The core setup phase will still wait for these tasks. If we don't want to wait for it we need to create it as a background task.

Reply:
Thank you very much for pointing this out! Will make a follow up PR.

Reply:
Follow up PR is here: https://github.com/home-assistant/core/pull/120985